forui/lib/src/widgets/pagination/pagination.dart

```diff
@@ -0,0 +1,238 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+
+/// A pagination enables the user to select a specific page from a range of pages.
+///
+/// See:
+/// * https://forui.dev/docs/navigation/pagination for working examples.
+/// * [FPaginationController] for customizing the pagination component's behavior.
+/// * [FPaginationStyle] for customizing the pagination component's appearance.
+final class FPagination extends StatefulWidget {
+  /// The pagination's style. Defaults to the appropriate style in [FThemeData.paginationStyle].
+  final FPaginationStyle? style;
+
+  /// The controller. Defaults to [FPaginationController.new].
+  final FPaginationController controller;
+
+  /// The previous button placed in the beginning of the pagination.
+  ///
+  /// Defaults to an `FAssets.icons.chevronLeft` icon.
+  final Widget? previous;
+
+  /// The next button placed at the end of the pagination.
+  ///
+  /// Defaults to an `FAssets.icons.chevronRight` icon.
+  final Widget? next;
+
+  /// Creates an [FPagination].
+  const FPagination({required this.controller, this.style, this.previous, this.next, super.key});
+
+  @override
+  State<FPagination> createState() => _FPaginationState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty('controller', controller));
+  }
+}
+
+class _FPaginationState extends State<FPagination> {
+  @override
+  void initState() {
+    super.initState();
+    widget.controller.addListener(() {
```

Comment:
For simplicity sake, I think it might make more sense to use a ValueListeneableBuilder widget or something similar. This way we can make management simpler.